src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -399,9 +399,26 @@ class CharacterPrediction implements IPrediction {
 		return cursor.clone().moveTo(this.appliedAt.pos) + input;
 	}
 
-	public matches(input: StringReader, lookBehind?: IPrediction) {
+	public matches(input: StringReader, lookBehind?: IPrediction, cursor?: Cursor) {
 		let startIndex = input.index;
 
+		const cursorMove = input.eatRe(/^\x1b\[(\d+);(\d+)H/);
```

Comment:
I might pop "move parsing" to its own function. This parses `H` which works for powershell today but we may need to parse other movement codes later..

Reply:
May also want to wrap it in a `while` in case there are multiple movements to restyle different parts of the text (which I think will always happen if the line is wrapped)

Reply:
ok moved it to its own function. Right now it still just handles H (aka CUP 🥤) but it loops. Any specific sequences you think we should handle?

Reply:
H is fine, we can add more as needed (e.g. C/D0